website/content/docs/job-specification/vault.mdx

```diff
@@ -109,6 +113,35 @@ vault {
 }
 ```
 
+### Private Token and Noop
+
+This example retrieves a Vault token that is not shared with the task when using
+a driver that provides `image` isolation like [Docker][docker].
+
+This allows to get a powerful Vault token that interacts with the task's
+[`template`][template] stanzas to issue all kinds of secrets (e.g., database
+secrets, other vault tokens, etc.), without sharing that issuing power with
+the task itself:
+
+```hcl
+vault {
+  policies = ["tls-policy", "nomad-job-policy"]
+  change_mode = "noop"
+  env = false
+  file = false
+}
+
+template {
+  data = <<-EOH
+  {{with secret "auth/token/create/nomad-job" "policies=examplepolicy"}}{{.Auth.ClientToken}}{{ end }}
+  EOH
+  destination = "${NOMAD_SECRETS_DIR}/examplepolicy.token"
+  change_mode = "noop"
```

Comment:
It's not clear from this example why we want `noop` here?

Reply:
Keeping it in the example but providing a little comment as to why you might want it seems fine.

I kind of like the notion of having it default to `noop` if both `env = false` and `file = false`, because all the client's internal "update" hooks would still get fired without touching the task ("noop" is unfortunately named, because it's not strictly no-op, just no-op for the task driver). Let's check that this wouldn't unexpectedly change the behavior of existing allocations after the client gets upgraded.

Reply:
@tgross 
> Let's check that this wouldn't unexpectedly change the behavior of existing allocations after the client gets upgraded.

I would assume that it's safe, since the `file` parameter
1. couldn't be specified beforehand (nomad doesn't accept unknown params in the vault stanza)
2. is new and defaults to true

so upgrading the client can never result in a situation where `env == true && file == true`.

Reply:
@tgross I extended the example to show how to use mix `change_mode`s and added a few words explaining what this means.